cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/sync/class-push-sync.php

```diff
@@ -389,30 +390,30 @@ public function prepare_upload( $post, $down_sync = false ) {
 			}
 
 			// If it's got a public ID, then this is an explicit update.
+			$settings   = $this->plugin->config['settings'];
 			$public_id  = $post->{Sync::META_KEYS['public_id']}; // use the __get method on the \WP_Post to get post_meta.
-			$dirs       = wp_get_upload_dir();
-			$cld_folder = false;
-			$folder     = trailingslashit( $dirs['cloudinary_folder'] );
-			if ( '/' === $dirs['cloudinary_folder'] ) {
-				$folder = '';
-			}
+			$cld_folder = trailingslashit( $settings['sync_media']['cloudinary_folder'] );
 			if ( empty( $public_id ) ) {
 				$file_info = pathinfo( $file );
-				$public_id = $folder . $file_info['filename'];
+				$public_id = $cld_folder . $file_info['filename'];
 			}
 
-			// Check if cloudinary folder is in public_id.
-			$parts = explode( '/', $public_id );
-			if ( untrailingslashit( $dirs['cloudinary_folder'] ) === $parts[0] ) {
-				$cld_folder = $dirs['cloudinary_folder'];
+			if ( false !== strpos( $public_id, '/' ) ) {
+				// Split the public_id into path and filename to allow filtering just the ID and not giving access to the path.
+				$public_id_info   = pathinfo( $public_id );
+				$public_id_folder = trailingslashit( $public_id_info['dirname'] );
+				$public_id_file   = $public_id_info['filename'];
+			} else {
+				// File is in the root of cloudinary.
+				$public_id_folder = '';
+				$public_id_file   = $public_id;
 			}
```

Comment:
Maybe we can switch to this in order to satisfy linters:

```suggestion
			// File is in the root of cloudinary.
			$public_id_folder = '';
			$public_id_file  = $public_id;
			
			if ( false !== strpos( $public_id, '/' ) ) {
				// Split the public_id into path and filename to allow filtering just the ID and not giving access to the path.
				$public_id_info   = pathinfo( $public_id );
				$public_id_folder = trailingslashit( $public_id_info['dirname'] );
				$public_id_file   = $public_id_info['filename'];
			} 
```

Reply:
@dugajean Nice idea. I usually do prefer to have a default, and if it's not what's expected then change.